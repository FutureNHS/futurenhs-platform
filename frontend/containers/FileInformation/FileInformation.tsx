import React, { FC } from "react";

import { format, parseISO } from "date-fns";
import Link from "next/link";
import styled from "styled-components";

import { EmailLink } from "../../components/EmailLink";
import { H2 } from "../../components/H2";
import { DownloadIcon, FileIcon } from "../../components/Icon";
import { MobileList, Table } from "../../components/Table";

// Hardcoded data until backend is done.
const fileVersions = [
  {
    id: "e92f20cb-58b0-464b-bbb3-7627c4b21628",
    title: "Version 1",
    createdAt: "2020-10-11 10:53:58.696734",
    emailAddress: "email@address.com",
    fileType: "csv",
  },
  {
    id: "55f4166f-af41-4513-a2aa-3b46519fb44e",
    title: "Version 2",
    createdAt: "2020-11-11 11:54:44.376148",
    emailAddress: "correct-grammar@address.com",
    fileType: "csv",
  },
  {
    id: "55f4166f-af41-4513-a2aa-3b46519fb44e",
    title: "Version 3",
    createdAt: "2020-12-11 19:54:44.376148",
    emailAddress: "perfectionist@address.com",
    fileType: "csv",
  },
];

const FileInfoLinks = styled.div`
  padding-bottom: 40px;
`;

interface FileVersion {
  id: string;
  title: string;
  createdAt: string;
  emailAddress: string;
  fileType: string;
}

const MobileTitle = styled.h3`
  font-size: 16px;
  font-weight: normal;
  margin: 0;
  padding-bottom: 20px;
`;

const CreatedDate = styled.span`
  color: ${({ theme }) => theme.colorNhsukGrey1};
`;

interface Props {
  workspaceId: string;
}

const FileInformation = ({ workspaceId }: Props) => {
  const previousVersions = fileVersions.slice(0, fileVersions.length - 1);
  const mobileTitleCell: FC<FileVersion> = ({ title }) => (
    <MobileTitle>{title}</MobileTitle>
  );

  const IconCell: FC<FileVersion> = ({ fileType }) => (
    <FileIcon fileType={fileType} />
  );

  const mobileEmailCell: FC<FileVersion> = ({ emailAddress }) => (
    <>
      <h4>Modified by</h4>
      <EmailLink emailAddress={emailAddress} />
    </>
  );

  const mobileCreatedAtCell: FC<FileVersion> = ({ createdAt }) => (
    <div>
      <h4>Modified on</h4>
      <p>{format(parseISO(createdAt), "d LLL yyyy kk:mm")}</p>
    </div>
  );

  const mobileDownloadCell: FC<FileVersion> = ({ id }) => (
    <Link href={`/workspaces/${workspaceId}/download/${id}`} passHref>
      <a>Download file</a>
    </Link>
  );

  const desktopTitleCell: FC<FileVersion> = ({ title }) => <>{title}</>;

  const desktopCreatedAtCell: FC<FileVersion> = ({ createdAt }) => (
    <CreatedDate>{format(parseISO(createdAt), "d LLL yyyy kk:mm")}</CreatedDate>
  );

  const desktopEmailCell: FC<FileVersion> = ({ emailAddress }) => (
    <EmailLink emailAddress={emailAddress} />
  );

  const desktopDownloadCell: FC<FileVersion> = ({ id }) => (
    <Link href={`/workspaces/${workspaceId}/download/${id}`} passHref>
      <a>
        <DownloadIcon />
      </a>
    </Link>
  );

  return (
    <>
      <H2 title="File information" />
      <h3>Contents</h3>
      <FileInfoLinks>
        <a>&#8212; {`Versions (${previousVersions.length}`})</a>
      </FileInfoLinks>

      <MobileList
        icon={IconCell}
        tableHeading="Previous versions"
        columns={[
          { content: mobileTitleCell },
          { content: mobileEmailCell },
          { content: mobileCreatedAtCell },
          { content: mobileDownloadCell },
        ]}
        data={previousVersions}
      />
      <Table
        icon={IconCell}
        dataCy="file-version-table"
        tableHeading="Previous versions"
        columns={[
          { heading: "Title", content: desktopTitleCell },
          { heading: "Modified by", content: desktopEmailCell },
          { heading: "Modified on", content: desktopCreatedAtCell },
          { content: desktopDownloadCell },
        ]}
        data={previousVersions}
      />
    </>
  );
};

export default FileInformation;
