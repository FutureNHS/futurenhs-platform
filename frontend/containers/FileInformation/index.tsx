import FileInformation from "./FileInformation";

export { FileInformation };
