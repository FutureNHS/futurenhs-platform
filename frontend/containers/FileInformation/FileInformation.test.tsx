import React from "react";

import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import theme from "../../lib/fixtures/theme.json";
import FileInformation from "./FileInformation";

describe(FileInformation, () => {
  it("renders as expected", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <FileInformation workspaceId="7cb46d97-beb4-4f48-a527-b7a1b0f9f219" />
      </ThemeProvider>
    );
    expect(container.asFragment()).toMatchSnapshot();
  });
});
