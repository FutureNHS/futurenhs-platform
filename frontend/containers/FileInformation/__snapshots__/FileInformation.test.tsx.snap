// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInformation renders as expected 1`] = `
<DocumentFragment>
  @media (min-width:641px) {

}

@media (min-width:641px) {

}

@media (min-width:641px) {

}

<h2>
    File information
  </h2>
  @media (min-width:641px) {

}

@media (min-width:641px) {

}

@media (min-width:641px) {

}

<h3>
    Contents
  </h3>
  .c0 {
  padding-bottom: 40px;
}

@media (min-width:641px) {

}

@media (min-width:641px) {

}

@media (min-width:641px) {

}

<div
    class="c0"
  >
    <a>
      â€” Versions (2)
    </a>
  </div>
  .c0 {
  background-color: #005eb8;
  color: rgb(255,255,255);
  display: inline-block;
  font-size: 1.5rem;
  line-height: 1.33333;
  padding: 8px 32px;
  position: relative;
  top: 34px;
}

@media (min-width:641px) {

}

@media (min-width:641px) {
  .c0 {
    display: none;
  }
}

@media (min-width:641px) {

}

<h3
    class="c0"
  >
    Previous versions
  </h3>
  .c1 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 26px;
  border-bottom: 1px solid rgb(216,221,224);
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 12px;
}

.c2 > div {
  padding-bottom: 10px;
}

.c2 p,
.c2 h4 {
  margin-bottom: 0;
  font-size: 16px;
}

.c2 p {
  color: rgb(76,98,114);
}

.c2 a {
  font-size: 16px;
}

.c0 {
  padding-top: 20px;
  padding-left: 12px;
  padding-right: 16px;
  background: rgb(240,244,245);
}

.c3 {
  font-size: 16px;
  font-weight: normal;
  margin: 0;
  padding-bottom: 20px;
}

@media (min-width:641px) {
  .c0 {
    display: none;
  }
}

@media (min-width:641px) {

}

@media (min-width:641px) {

}

<ul
    class="c0"
  >
    <li
      class="c1"
    >
      <div
        class="icon-wrapper file-icon-wrapper "
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 28 28"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M22.5866 25.6666H5.41329C5.00092 25.6666 4.66663 25.3324 4.66663 24.92V3.07998C4.66663 2.66761 5.00092 2.33331 5.41329 2.33331H18.1766V6.58651C18.1766 7.08548 18.5811 7.48998 19.0801 7.48998H23.3333V24.92C23.3333 25.3324 22.999 25.6666 22.5866 25.6666ZM18.6666 6.25331V2.33331L23.3333 6.99998H19.4133C19.0009 6.99998 18.6666 6.66569 18.6666 6.25331ZM12.8346 8.16668H15.9916V9.67868H12.8346V8.16668ZM11.3237 8.16669H8.16674V9.67869H11.3237V8.16669ZM12.2031 20.6386L11.5164 20.4209C11.4574 20.6777 11.3601 20.8659 11.2245 20.9855C11.0889 21.1052 10.927 21.165 10.7388 21.165C10.4836 21.165 10.2763 21.0709 10.1168 20.8827C9.95725 20.6944 9.8775 20.3786 9.8775 19.9352C9.8775 19.5173 9.95805 19.215 10.1192 19.0284C10.2818 18.8418 10.4932 18.7485 10.7532 18.7485C10.9414 18.7485 11.1009 18.8011 11.2317 18.9064C11.3641 19.0117 11.451 19.1552 11.4925 19.3371L12.1935 19.1696C12.1138 18.8888 11.9941 18.6735 11.8346 18.5236C11.5667 18.27 11.2181 18.1432 10.7891 18.1432C10.2978 18.1432 9.90143 18.3051 9.59996 18.6289C9.2985 18.9511 9.14777 19.404 9.14777 19.9878C9.14777 20.5397 9.2977 20.9752 9.59757 21.2942C9.89744 21.6116 10.2803 21.7703 10.746 21.7703C11.1224 21.7703 11.4327 21.6778 11.6767 21.4928C11.9224 21.3061 12.0978 21.0214 12.2031 20.6386ZM13.3228 20.5022L12.6338 20.5692C12.68 20.9616 12.822 21.2607 13.0596 21.4665C13.2973 21.6706 13.6378 21.7727 14.0813 21.7727C14.3859 21.7727 14.6403 21.7304 14.8445 21.6459C15.0487 21.5598 15.2066 21.429 15.3182 21.2535C15.4299 21.0781 15.4857 20.8898 15.4857 20.6889C15.4857 20.4672 15.4387 20.2813 15.3445 20.1314C15.252 19.9799 15.1228 19.861 14.9569 19.7749C14.7927 19.6872 14.5382 19.6026 14.1937 19.5213C13.8492 19.4399 13.6323 19.3618 13.5429 19.2868C13.4727 19.2278 13.4377 19.1568 13.4377 19.0739C13.4377 18.983 13.4751 18.9104 13.5501 18.8561C13.6665 18.7716 13.8276 18.7293 14.0334 18.7293C14.2328 18.7293 14.3819 18.7692 14.4808 18.849C14.5813 18.9271 14.6467 19.0563 14.677 19.2366L15.3852 19.2055C15.3741 18.8833 15.2568 18.6257 15.0335 18.4327C14.8118 18.2397 14.4808 18.1432 14.0406 18.1432C13.771 18.1432 13.5405 18.1838 13.3491 18.2652C13.1593 18.3465 13.0134 18.4654 12.9113 18.6217C12.8108 18.7764 12.7606 18.9431 12.7606 19.1217C12.7606 19.3993 12.8682 19.6345 13.0836 19.8275C13.2367 19.9647 13.5031 20.0804 13.8827 20.1745C14.1778 20.2478 14.3668 20.2989 14.4497 20.3276C14.5709 20.3707 14.6555 20.4217 14.7033 20.4807C14.7528 20.5381 14.7775 20.6083 14.7775 20.6913C14.7775 20.8205 14.7193 20.9337 14.6028 21.031C14.488 21.1267 14.3165 21.1746 14.0884 21.1746C13.8731 21.1746 13.7016 21.1203 13.574 21.0119C13.448 20.9034 13.3643 20.7335 13.3228 20.5022ZM16.9763 21.7105L15.7226 18.203H16.4906L17.3782 20.7989L18.2372 18.203H18.9884L17.7323 21.7105H16.9763ZM8.21102 10.9678H11.3237V12.4798H8.21102V10.9678ZM11.3237 13.8133H8.21102V15.3253H11.3237V13.8133ZM15.9915 10.9678H12.8788V12.4798H15.9915V10.9678ZM12.8788 13.8133H15.9915V15.3253H12.8788V13.8133ZM20.6607 10.9678H17.548V12.4798H20.6607V10.9678ZM17.548 13.8133H20.6607V15.3253H17.548V13.8133ZM20.6595 9.67868H17.5036V8.16668H18.2876L20.6595 9.67868Z"
            fill="#007F3B"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="c2"
      >
        <div>
          <h3
            class="c3"
          >
            Version 1
          </h3>
        </div>
        <div>
          <h4>
            Modified by
          </h4>
          <a
            href="mailto:email@address.com"
            rel="noreferrer"
            target="_blank"
          >
            email@address.com
          </a>
        </div>
        <div>
          <div>
            <h4>
              Modified on
            </h4>
            <p>
              11 Oct 2020 10:53
            </p>
          </div>
        </div>
        <div>
          <a
            href="/workspaces/7cb46d97-beb4-4f48-a527-b7a1b0f9f219/download/e92f20cb-58b0-464b-bbb3-7627c4b21628"
          >
            Download file
          </a>
        </div>
      </div>
    </li>
    <li
      class="c1"
    >
      <div
        class="icon-wrapper file-icon-wrapper "
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 28 28"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M22.5866 25.6666H5.41329C5.00092 25.6666 4.66663 25.3324 4.66663 24.92V3.07998C4.66663 2.66761 5.00092 2.33331 5.41329 2.33331H18.1766V6.58651C18.1766 7.08548 18.5811 7.48998 19.0801 7.48998H23.3333V24.92C23.3333 25.3324 22.999 25.6666 22.5866 25.6666ZM18.6666 6.25331V2.33331L23.3333 6.99998H19.4133C19.0009 6.99998 18.6666 6.66569 18.6666 6.25331ZM12.8346 8.16668H15.9916V9.67868H12.8346V8.16668ZM11.3237 8.16669H8.16674V9.67869H11.3237V8.16669ZM12.2031 20.6386L11.5164 20.4209C11.4574 20.6777 11.3601 20.8659 11.2245 20.9855C11.0889 21.1052 10.927 21.165 10.7388 21.165C10.4836 21.165 10.2763 21.0709 10.1168 20.8827C9.95725 20.6944 9.8775 20.3786 9.8775 19.9352C9.8775 19.5173 9.95805 19.215 10.1192 19.0284C10.2818 18.8418 10.4932 18.7485 10.7532 18.7485C10.9414 18.7485 11.1009 18.8011 11.2317 18.9064C11.3641 19.0117 11.451 19.1552 11.4925 19.3371L12.1935 19.1696C12.1138 18.8888 11.9941 18.6735 11.8346 18.5236C11.5667 18.27 11.2181 18.1432 10.7891 18.1432C10.2978 18.1432 9.90143 18.3051 9.59996 18.6289C9.2985 18.9511 9.14777 19.404 9.14777 19.9878C9.14777 20.5397 9.2977 20.9752 9.59757 21.2942C9.89744 21.6116 10.2803 21.7703 10.746 21.7703C11.1224 21.7703 11.4327 21.6778 11.6767 21.4928C11.9224 21.3061 12.0978 21.0214 12.2031 20.6386ZM13.3228 20.5022L12.6338 20.5692C12.68 20.9616 12.822 21.2607 13.0596 21.4665C13.2973 21.6706 13.6378 21.7727 14.0813 21.7727C14.3859 21.7727 14.6403 21.7304 14.8445 21.6459C15.0487 21.5598 15.2066 21.429 15.3182 21.2535C15.4299 21.0781 15.4857 20.8898 15.4857 20.6889C15.4857 20.4672 15.4387 20.2813 15.3445 20.1314C15.252 19.9799 15.1228 19.861 14.9569 19.7749C14.7927 19.6872 14.5382 19.6026 14.1937 19.5213C13.8492 19.4399 13.6323 19.3618 13.5429 19.2868C13.4727 19.2278 13.4377 19.1568 13.4377 19.0739C13.4377 18.983 13.4751 18.9104 13.5501 18.8561C13.6665 18.7716 13.8276 18.7293 14.0334 18.7293C14.2328 18.7293 14.3819 18.7692 14.4808 18.849C14.5813 18.9271 14.6467 19.0563 14.677 19.2366L15.3852 19.2055C15.3741 18.8833 15.2568 18.6257 15.0335 18.4327C14.8118 18.2397 14.4808 18.1432 14.0406 18.1432C13.771 18.1432 13.5405 18.1838 13.3491 18.2652C13.1593 18.3465 13.0134 18.4654 12.9113 18.6217C12.8108 18.7764 12.7606 18.9431 12.7606 19.1217C12.7606 19.3993 12.8682 19.6345 13.0836 19.8275C13.2367 19.9647 13.5031 20.0804 13.8827 20.1745C14.1778 20.2478 14.3668 20.2989 14.4497 20.3276C14.5709 20.3707 14.6555 20.4217 14.7033 20.4807C14.7528 20.5381 14.7775 20.6083 14.7775 20.6913C14.7775 20.8205 14.7193 20.9337 14.6028 21.031C14.488 21.1267 14.3165 21.1746 14.0884 21.1746C13.8731 21.1746 13.7016 21.1203 13.574 21.0119C13.448 20.9034 13.3643 20.7335 13.3228 20.5022ZM16.9763 21.7105L15.7226 18.203H16.4906L17.3782 20.7989L18.2372 18.203H18.9884L17.7323 21.7105H16.9763ZM8.21102 10.9678H11.3237V12.4798H8.21102V10.9678ZM11.3237 13.8133H8.21102V15.3253H11.3237V13.8133ZM15.9915 10.9678H12.8788V12.4798H15.9915V10.9678ZM12.8788 13.8133H15.9915V15.3253H12.8788V13.8133ZM20.6607 10.9678H17.548V12.4798H20.6607V10.9678ZM17.548 13.8133H20.6607V15.3253H17.548V13.8133ZM20.6595 9.67868H17.5036V8.16668H18.2876L20.6595 9.67868Z"
            fill="#007F3B"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="c2"
      >
        <div>
          <h3
            class="c3"
          >
            Version 2
          </h3>
        </div>
        <div>
          <h4>
            Modified by
          </h4>
          <a
            href="mailto:correct-grammar@address.com"
            rel="noreferrer"
            target="_blank"
          >
            correct-grammar@address.com
          </a>
        </div>
        <div>
          <div>
            <h4>
              Modified on
            </h4>
            <p>
              11 Nov 2020 11:54
            </p>
          </div>
        </div>
        <div>
          <a
            href="/workspaces/7cb46d97-beb4-4f48-a527-b7a1b0f9f219/download/55f4166f-af41-4513-a2aa-3b46519fb44e"
          >
            Download file
          </a>
        </div>
      </div>
    </li>
  </ul>
  .c0 {
  display: none;
  width: 100%;
}

.c0 > div {
  background: rgb(240,244,245);
}

.c0.have-expander table {
  table-layout: fixed;
  border-collapse: collapse;
  width: 100%;
}

.c0.have-expander th:first-child {
  width: 30%;
}

.c0.have-expander th:last-child {
  width: 10%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 .file-icon-wrapper {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-right: 8px;
}

tbody tr:hover:not(.c1.withoutBottomBorder) {
  background: rgb(255,255,255);
}

.c2 td {
  font-size: 16px;
}

.c3 a[href^="mailto"] {
  word-break: break-all;
}

.c3.withoutBottomBorder {
  border-bottom: none;
}

.c5 {
  color: rgb(76,98,114);
}

@media (min-width:641px) {

}

@media (min-width:641px) {

}

@media (min-width:641px) {
  .c0 {
    display: block;
  }
}

<div
    class="c0"
  >
    <div
      class="nhsuk-table__panel-with-heading-tab"
    >
      <h3
        class="nhsuk-table__heading-tab"
      >
        Previous versions
      </h3>
      <div
        class="nhsuk-table-responsive"
      >
        <table
          class="nhsuk-table c1 c2"
          data-cy="file-version-table"
        >
          <thead
            class="nhsuk-table__head"
          >
            <tr
              class="nhsuk-table__row"
            >
              <th
                class="nhsuk-table__header"
                scope="col"
              >
                Title
              </th>
              <th
                class="nhsuk-table__header"
                scope="col"
              >
                Modified by
              </th>
              <th
                class="nhsuk-table__header"
                scope="col"
              >
                Modified on
              </th>
              <th
                class="nhsuk-table__header"
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="nhsuk-table__body"
          >
            <tr
              class="nhsuk-table__row"
            >
              <td
                class="nhsuk-table__cell c3"
              >
                <div
                  class="c4"
                >
                  <div
                    class="icon-wrapper file-icon-wrapper "
                  >
                    <svg
                      fill="none"
                      height="28"
                      viewBox="0 0 28 28"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M22.5866 25.6666H5.41329C5.00092 25.6666 4.66663 25.3324 4.66663 24.92V3.07998C4.66663 2.66761 5.00092 2.33331 5.41329 2.33331H18.1766V6.58651C18.1766 7.08548 18.5811 7.48998 19.0801 7.48998H23.3333V24.92C23.3333 25.3324 22.999 25.6666 22.5866 25.6666ZM18.6666 6.25331V2.33331L23.3333 6.99998H19.4133C19.0009 6.99998 18.6666 6.66569 18.6666 6.25331ZM12.8346 8.16668H15.9916V9.67868H12.8346V8.16668ZM11.3237 8.16669H8.16674V9.67869H11.3237V8.16669ZM12.2031 20.6386L11.5164 20.4209C11.4574 20.6777 11.3601 20.8659 11.2245 20.9855C11.0889 21.1052 10.927 21.165 10.7388 21.165C10.4836 21.165 10.2763 21.0709 10.1168 20.8827C9.95725 20.6944 9.8775 20.3786 9.8775 19.9352C9.8775 19.5173 9.95805 19.215 10.1192 19.0284C10.2818 18.8418 10.4932 18.7485 10.7532 18.7485C10.9414 18.7485 11.1009 18.8011 11.2317 18.9064C11.3641 19.0117 11.451 19.1552 11.4925 19.3371L12.1935 19.1696C12.1138 18.8888 11.9941 18.6735 11.8346 18.5236C11.5667 18.27 11.2181 18.1432 10.7891 18.1432C10.2978 18.1432 9.90143 18.3051 9.59996 18.6289C9.2985 18.9511 9.14777 19.404 9.14777 19.9878C9.14777 20.5397 9.2977 20.9752 9.59757 21.2942C9.89744 21.6116 10.2803 21.7703 10.746 21.7703C11.1224 21.7703 11.4327 21.6778 11.6767 21.4928C11.9224 21.3061 12.0978 21.0214 12.2031 20.6386ZM13.3228 20.5022L12.6338 20.5692C12.68 20.9616 12.822 21.2607 13.0596 21.4665C13.2973 21.6706 13.6378 21.7727 14.0813 21.7727C14.3859 21.7727 14.6403 21.7304 14.8445 21.6459C15.0487 21.5598 15.2066 21.429 15.3182 21.2535C15.4299 21.0781 15.4857 20.8898 15.4857 20.6889C15.4857 20.4672 15.4387 20.2813 15.3445 20.1314C15.252 19.9799 15.1228 19.861 14.9569 19.7749C14.7927 19.6872 14.5382 19.6026 14.1937 19.5213C13.8492 19.4399 13.6323 19.3618 13.5429 19.2868C13.4727 19.2278 13.4377 19.1568 13.4377 19.0739C13.4377 18.983 13.4751 18.9104 13.5501 18.8561C13.6665 18.7716 13.8276 18.7293 14.0334 18.7293C14.2328 18.7293 14.3819 18.7692 14.4808 18.849C14.5813 18.9271 14.6467 19.0563 14.677 19.2366L15.3852 19.2055C15.3741 18.8833 15.2568 18.6257 15.0335 18.4327C14.8118 18.2397 14.4808 18.1432 14.0406 18.1432C13.771 18.1432 13.5405 18.1838 13.3491 18.2652C13.1593 18.3465 13.0134 18.4654 12.9113 18.6217C12.8108 18.7764 12.7606 18.9431 12.7606 19.1217C12.7606 19.3993 12.8682 19.6345 13.0836 19.8275C13.2367 19.9647 13.5031 20.0804 13.8827 20.1745C14.1778 20.2478 14.3668 20.2989 14.4497 20.3276C14.5709 20.3707 14.6555 20.4217 14.7033 20.4807C14.7528 20.5381 14.7775 20.6083 14.7775 20.6913C14.7775 20.8205 14.7193 20.9337 14.6028 21.031C14.488 21.1267 14.3165 21.1746 14.0884 21.1746C13.8731 21.1746 13.7016 21.1203 13.574 21.0119C13.448 20.9034 13.3643 20.7335 13.3228 20.5022ZM16.9763 21.7105L15.7226 18.203H16.4906L17.3782 20.7989L18.2372 18.203H18.9884L17.7323 21.7105H16.9763ZM8.21102 10.9678H11.3237V12.4798H8.21102V10.9678ZM11.3237 13.8133H8.21102V15.3253H11.3237V13.8133ZM15.9915 10.9678H12.8788V12.4798H15.9915V10.9678ZM12.8788 13.8133H15.9915V15.3253H12.8788V13.8133ZM20.6607 10.9678H17.548V12.4798H20.6607V10.9678ZM17.548 13.8133H20.6607V15.3253H17.548V13.8133ZM20.6595 9.67868H17.5036V8.16668H18.2876L20.6595 9.67868Z"
                        fill="#007F3B"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  Version 1
                </div>
              </td>
              <td
                class="nhsuk-table__cell c3"
              >
                <a
                  href="mailto:email@address.com"
                  rel="noreferrer"
                  target="_blank"
                >
                  email@address.com
                </a>
              </td>
              <td
                class="nhsuk-table__cell c3"
              >
                <span
                  class="c5"
                >
                  11 Oct 2020 10:53
                </span>
              </td>
              <td
                class="nhsuk-table__cell c3"
              >
                <a
                  href="/workspaces/7cb46d97-beb4-4f48-a527-b7a1b0f9f219/download/e92f20cb-58b0-464b-bbb3-7627c4b21628"
                >
                  <div
                    class="icon-wrapper "
                  >
                    <svg
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M20 19.6923V22H4V19.6923H20ZM13.1443 2V13.3462L16.75 9.8125L18.2955 11.3269L12.1136 17.3846L5.93182 11.3269L7.47727 9.8125L11.083 13.3462V2H13.1443Z"
                        fill="#005EB8"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </a>
              </td>
            </tr>
            <tr
              class="nhsuk-table__row"
            >
              <td
                class="nhsuk-table__cell c3"
              >
                <div
                  class="c4"
                >
                  <div
                    class="icon-wrapper file-icon-wrapper "
                  >
                    <svg
                      fill="none"
                      height="28"
                      viewBox="0 0 28 28"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M22.5866 25.6666H5.41329C5.00092 25.6666 4.66663 25.3324 4.66663 24.92V3.07998C4.66663 2.66761 5.00092 2.33331 5.41329 2.33331H18.1766V6.58651C18.1766 7.08548 18.5811 7.48998 19.0801 7.48998H23.3333V24.92C23.3333 25.3324 22.999 25.6666 22.5866 25.6666ZM18.6666 6.25331V2.33331L23.3333 6.99998H19.4133C19.0009 6.99998 18.6666 6.66569 18.6666 6.25331ZM12.8346 8.16668H15.9916V9.67868H12.8346V8.16668ZM11.3237 8.16669H8.16674V9.67869H11.3237V8.16669ZM12.2031 20.6386L11.5164 20.4209C11.4574 20.6777 11.3601 20.8659 11.2245 20.9855C11.0889 21.1052 10.927 21.165 10.7388 21.165C10.4836 21.165 10.2763 21.0709 10.1168 20.8827C9.95725 20.6944 9.8775 20.3786 9.8775 19.9352C9.8775 19.5173 9.95805 19.215 10.1192 19.0284C10.2818 18.8418 10.4932 18.7485 10.7532 18.7485C10.9414 18.7485 11.1009 18.8011 11.2317 18.9064C11.3641 19.0117 11.451 19.1552 11.4925 19.3371L12.1935 19.1696C12.1138 18.8888 11.9941 18.6735 11.8346 18.5236C11.5667 18.27 11.2181 18.1432 10.7891 18.1432C10.2978 18.1432 9.90143 18.3051 9.59996 18.6289C9.2985 18.9511 9.14777 19.404 9.14777 19.9878C9.14777 20.5397 9.2977 20.9752 9.59757 21.2942C9.89744 21.6116 10.2803 21.7703 10.746 21.7703C11.1224 21.7703 11.4327 21.6778 11.6767 21.4928C11.9224 21.3061 12.0978 21.0214 12.2031 20.6386ZM13.3228 20.5022L12.6338 20.5692C12.68 20.9616 12.822 21.2607 13.0596 21.4665C13.2973 21.6706 13.6378 21.7727 14.0813 21.7727C14.3859 21.7727 14.6403 21.7304 14.8445 21.6459C15.0487 21.5598 15.2066 21.429 15.3182 21.2535C15.4299 21.0781 15.4857 20.8898 15.4857 20.6889C15.4857 20.4672 15.4387 20.2813 15.3445 20.1314C15.252 19.9799 15.1228 19.861 14.9569 19.7749C14.7927 19.6872 14.5382 19.6026 14.1937 19.5213C13.8492 19.4399 13.6323 19.3618 13.5429 19.2868C13.4727 19.2278 13.4377 19.1568 13.4377 19.0739C13.4377 18.983 13.4751 18.9104 13.5501 18.8561C13.6665 18.7716 13.8276 18.7293 14.0334 18.7293C14.2328 18.7293 14.3819 18.7692 14.4808 18.849C14.5813 18.9271 14.6467 19.0563 14.677 19.2366L15.3852 19.2055C15.3741 18.8833 15.2568 18.6257 15.0335 18.4327C14.8118 18.2397 14.4808 18.1432 14.0406 18.1432C13.771 18.1432 13.5405 18.1838 13.3491 18.2652C13.1593 18.3465 13.0134 18.4654 12.9113 18.6217C12.8108 18.7764 12.7606 18.9431 12.7606 19.1217C12.7606 19.3993 12.8682 19.6345 13.0836 19.8275C13.2367 19.9647 13.5031 20.0804 13.8827 20.1745C14.1778 20.2478 14.3668 20.2989 14.4497 20.3276C14.5709 20.3707 14.6555 20.4217 14.7033 20.4807C14.7528 20.5381 14.7775 20.6083 14.7775 20.6913C14.7775 20.8205 14.7193 20.9337 14.6028 21.031C14.488 21.1267 14.3165 21.1746 14.0884 21.1746C13.8731 21.1746 13.7016 21.1203 13.574 21.0119C13.448 20.9034 13.3643 20.7335 13.3228 20.5022ZM16.9763 21.7105L15.7226 18.203H16.4906L17.3782 20.7989L18.2372 18.203H18.9884L17.7323 21.7105H16.9763ZM8.21102 10.9678H11.3237V12.4798H8.21102V10.9678ZM11.3237 13.8133H8.21102V15.3253H11.3237V13.8133ZM15.9915 10.9678H12.8788V12.4798H15.9915V10.9678ZM12.8788 13.8133H15.9915V15.3253H12.8788V13.8133ZM20.6607 10.9678H17.548V12.4798H20.6607V10.9678ZM17.548 13.8133H20.6607V15.3253H17.548V13.8133ZM20.6595 9.67868H17.5036V8.16668H18.2876L20.6595 9.67868Z"
                        fill="#007F3B"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  Version 2
                </div>
              </td>
              <td
                class="nhsuk-table__cell c3"
              >
                <a
                  href="mailto:correct-grammar@address.com"
                  rel="noreferrer"
                  target="_blank"
                >
                  correct-grammar@address.com
                </a>
              </td>
              <td
                class="nhsuk-table__cell c3"
              >
                <span
                  class="c5"
                >
                  11 Nov 2020 11:54
                </span>
              </td>
              <td
                class="nhsuk-table__cell c3"
              >
                <a
                  href="/workspaces/7cb46d97-beb4-4f48-a527-b7a1b0f9f219/download/55f4166f-af41-4513-a2aa-3b46519fb44e"
                >
                  <div
                    class="icon-wrapper "
                  >
                    <svg
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M20 19.6923V22H4V19.6923H20ZM13.1443 2V13.3462L16.75 9.8125L18.2955 11.3269L12.1136 17.3846L5.93182 11.3269L7.47727 9.8125L11.083 13.3462V2H13.1443Z"
                        fill="#005EB8"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
